### PR TITLE
Return weights in default weight unit

### LINKED ISSUE #5874: Product type shipping weight gets saved in kilograms when set to pounds
### What I'm trying to achieve
Shipping weight unit is set to POUNDS and i am trying to configure weight on product types in pounds. When saved and refreshed it gets converted to kilograms.

### Steps to reproduce the problem
1. Configuration -> Shipping Methods -> Set shipping weight unit to LB
2. Go to product type that is shippable and add weight in pounds
3. Refresh page

### What I expected to happen
I expect the weight to be shown and used in pounds when set to pounds.

### Screenshots

Step 1
shipping-weight

Step 2
product-type-weight-unsaved

Step 3
product-type-weight-saved

### LINKED ISSUE #5875: Shipping rate is converted to KG when set to different shipping weight units
### What I'm trying to achieve
I want to set the shipping weight unit to pounds and be able to configure the weight based rates in pounds.

### Steps to reproduce the problem
1. Configuration -> Shipping Methods -> Set shipping weight unit to LB
2.  Create a shipping zone
3. Create a weight based rate using pounds as unit type

### What I expected to happen
I expected the weight range to show up in pounds.

### Screenshots
shipping-weight

edit-weight

weight-based-rate

### PR DESCRIPTION
**Pull request by IKarbowiak**
Weights should be returned in shop default unit weights.
Fixes #5874
Fixes #5875 

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [x] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

image
image
image

### CODE REVIEW

File: saleor/core/weight.py [hunk 1]
```diff
@@ -35,13 +35,26 @@ def zero_weight():
     return Weight(kg=0)
 
 
-def convert_weight(weight, unit):
+def convert_weight(weight: Weight, unit: str) -> Weight:
```
Comment (fowczarek) [inline saleor/core/weight.py:38]: Maby better to use `WeightUnits` instead of `str` as `unit` type?

File: saleor/core/weight.py [hunk 2]
```diff
@@ -35,13 +35,26 @@ def zero_weight():
...
+    weight.value = round(weight.value, 3)
+    return weight
 
 
 def get_default_weight_unit():
     site = Site.objects.get_current()
     return site.settings.default_weight_unit
+
+
+def covert_weight_to_default_weight_unit(weight: Weight) -> Weight:
```
Comment (maarcingebala) [inline saleor/core/weight.py:52]: Typo: covert :)

File: saleor/core/weight.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/core/weight.py:38]: Would be nice if we could use enum here.

Comment (maarcingebala) [reply saleor/core/weight.py:38]: Also let's add a docstring here, even very simple that it just coverts weight to given unit.

Comment (IKarbowiak) [reply saleor/core/weight.py:38]: There is a short description in a comment. I can covert it to the docstring

Comment (IKarbowiak) [reply saleor/core/weight.py:38]: Must be `str` because `getattr` require `str`.

**Pull request merged**